### DISCUSSION TITLE
Building Custom Docker Image

### DISCUSSION THREAD

**Discussion by RafaelCRC** [gravitational/teleport#38397] [Q&A] [answered]
Hello everyone. I am currently working on a project that involves extending the functionality of teleport by incorporating custom code. However, I've encountered challenges while trying to build a custom Docker image with the Teleport repository.

After some research, I couldn't find any official documentation on the recommended process for building a custom Docker image. 

Could someone from the community or the maintainers provide guidance on how to achieve this?

**Comment by webvictim** [comment]
Create a `Dockerfile`
```
FROM public.ecr.aws/gravitational/teleport-distroless:15.0.2
# add other Docker commands you want here
```

Then `docker build -t <tag> .`

If you need a shell in your image, use `public.ecr.aws/gravitational/teleport-distroless-debug:15.0.2` instead.

**Reply by RafaelCRC** [reply]
Thanks for the quick response, but what I want is how to build an image equivalent to this one, I saw that in the Docker documentation there is the "make build" command, but it is only used for testing according to what is written.

**Reply by webvictim** [reply, marked as answer]
Take a look at these Dockerfiles:

- https://github.com/gravitational/teleport/blob/master/build.assets/charts/Dockerfile-distroless
- https://github.com/gravitational/teleport/blob/master/build.assets/charts/Dockerfile
